astropy/table/tests/test_pprint.py

```diff
@@ -917,3 +917,23 @@ def test_output_globs(self):
         with t.pprint_exclude_names.set('a??'):
             out = t.pformat_all()
         assert out == exp
+
+
+def test_repr_latex():
+    """
+    Tests the _repr_latex_ method of astropy.table.Table class.
+    Output should be a latex formatted table printed on the screen.
+    Test for PR related to issue #3972.
+    """
+    from astropy.io import ascii
+    a = [1, 2, 3, 5]
+    b = np.array([4, 5, 6, 5])
+    c = ['hello', 'konichiwa', 'namaskar', 'guten morgen']
+    d = [2, 3, 5, 10] * u.m/u.s**2
+    t = Table([a, b, c, d], names=('val', 'val1', 'greeting', 'speed'))
+
+    t._repr_latex_(latexdict='AA')
+    t._repr_latex_(latexdict='doublelines')
+    t._repr_latex_(latexdict='template')
+    t._repr_latex_(latexdict={'caption':'demo table'})
```

Comment:
It's OK for you to print something on a screen to test, but before we merge, we must make that test automatic. Astropy has > 10000 tests and nobody want to look at that many output every time to see tat it still does what it should. Since (see above) the function should return a long string, you can use strong comparison and hardcode the exact output you expect. Something like:
```
assert t._repr_latex_(...) == '''
\hline
hello & [whatever the output should be here]
```
It's probably enough to test just two examples with two columns, if you want it easier, but I like this table, too!